src/main/scala/org/citegraph/Application.scala

```diff
@@ -13,18 +20,51 @@ object Application {
     println("USAGE")
     println("\tBuild:\n\t\tsbt package")
     println("\tSubmit as JAR to Spark cluster:\n\t\t$SPARK_HOME/bin/spark-submit <submit_options> \\")
-    println("\t\ttarget/scala-2.13/citegraph_2.13-0.1.jar <hdfs_file>")
-    println("\t\trequires option csv_directory needs to have path to where the data csv files are")
+    println("\t\ttarget/scala-2.13/citegraph_2.13-0.1.jar <hdfs_input_dir> <hdfs_output_dir>")
     println()
   }
 
+  def isValidHdfsUri(uri: String): Boolean = {
+    uri.startsWith("hdfs://")
+  }
+
   def main(args: Array[String]): Unit = {
     printArgs(args)
     if (args.length != 2) {
       printUsage()
       System.exit(1)
     }
 
+    // Validate and fix inputDirectory
+    var inputDirectory: String = args(0)
+    val outputDirectory: String = args(1)
+    if (!isValidHdfsUri(inputDirectory)) {
+      printf("Invalid HDFS input directory: %s\n", inputDirectory)
+      System.exit(1)
+    } else if (!isValidHdfsUri(outputDirectory)) {
+      printf("Invalid HDFS output directory: %s\n", outputDirectory)
+      System.exit(1)
+    }
+
+    if (inputDirectory.endsWith("/")) { // Chop off trailing slash
+      inputDirectory = inputDirectory.substring(0, inputDirectory.length - 1)
+    }
+    printf("inputDirectory: %s, outputDirectory: %s\n", inputDirectory, outputDirectory)
+
+    // Make a SparkSession and SparkContext
+    val sparkSession: SparkSession = SparkSession.builder
+      .appName("CiteGraph")
+      .getOrCreate()
+
+
+    val dataframeLoader: DataFrameLoader = new DataFrameLoader(inputDirectory, sparkSession)
+    val citationsDF: DataFrame = dataframeLoader.loadCitations()
+    val publishedDatesDF: DataFrame = dataframeLoader.loadPublishedDates()
+
+    val analytics: Analytics = new Analytics(sparkSession, citationsDF, publishedDatesDF)
+    analytics.findDensitiesByYear()
+
+    sparkSession.close()
```

Comment:
This is very clean, concise and easy to read. Nicely done.

Reply:
Thanks!